src/wp-includes/block-supports/layout.php

```diff
@@ -230,4 +250,65 @@ static function( $matches ) {
 	return $updated_content;
 }
 
-add_filter( 'render_block', 'wp_restore_group_inner_container', 10, 2 );
+add_filter( 'render_block_core/group', 'wp_restore_group_inner_container', 10, 2 );
+
+/**
+ * For themes without theme.json file, make sure
+ * to restore the outer div for the aligned image block
+ * to avoid breaking styles relying on that div.
+ *
+ * @since 6.0.0
+ * @access private
+ *
+ * @param string $block_content Rendered block content.
+ * @param  array  $block        Block object.
+ * @return string Filtered block content.
+ */
+function wp_restore_image_outer_container( $block_content, $block ) {
+	$image_with_align = "
+/# 1) everything up to the class attribute contents
+(
+	^\s*
+	<figure\b
+	[^>]*
+	\bclass=
+	[\"']
+)
+# 2) the class attribute contents
+(
+	[^\"']*
+	\bwp-block-image\b
+	[^\"']*
+	\b(?:alignleft|alignright|aligncenter)\b
+	[^\"']*
+)
+# 3) everything after the class attribute contents
+(
+	[\"']
+	[^>]*
+	>
+	.*
+	<\/figure>
+)/iUx";
+
+	if (
+		WP_Theme_JSON_Resolver::theme_has_support() ||
+		0 === preg_match( $image_with_align, $block_content, $matches )
+	) {
+		return $block_content;
+	}
+
+	$wrapper_classnames = array( 'wp-block-image' );
+
+	// If the block has a classNames attribute these classnames need to be removed from the content and added back
+	// to the new wrapper div also.
+	if ( ! empty( $block['attrs']['className'] ) ) {
+		$wrapper_classnames = array_merge( $wrapper_classnames, explode( ' ', $block['attrs']['className'] ) );
+	}
+	$content_classnames          = explode( ' ', $matches[2] );
+	$filtered_content_classnames = array_diff( $content_classnames, $wrapper_classnames );
+
+	return '<div class="' . implode( ' ', $wrapper_classnames ) . '">' . $matches[1] . implode( ' ', $filtered_content_classnames ) . $matches[3] . '</div>';
+}
+
+add_filter( 'render_block_core/image', 'wp_restore_image_outer_container', 10, 2 );
```

Comment:
cc @glendaviesnz and @youknowriad to check the migration on this one. Thanks!

Reply:
This is looking good to me.